x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java

```diff
@@ -254,7 +255,7 @@ public void testCreateAndRestoreSearchableSnapshot() throws Exception {
         assertSearchableSnapshotStats(restoredIndexName, cacheEnabled, nonCachedExtensions);
 
         ensureGreen(restoredIndexName);
-        assertBusy(() -> assertShardFolders(restoredIndexName, true));
+        assertBusy(() -> assertShardFolders(restoredIndexName, true), 30, TimeUnit.SECONDS);
```

Comment:
This is consistent with another change: https://github.com/elastic/elasticsearch/pull/84942

Reply:
It looks like this reproduces only if we close the index (eg index files are kept on the disk) **and** we reuse the same index name for the mounted index. This is likely resulting in files not cleaned in time.